torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -2041,12 +2037,25 @@ std::vector<Val*> Index::getGlobalConsumerStridedIndices(
           strides[dim], getHaloExtentOfRootAxis(root_dom[dim]));
     }
   }
+  return strides;
+}
 
-  auto vectorize_shift =
-      loops.empty() ? nullptr : loops.back()->vectorize_shift();
+std::vector<Val*> Index::getRootIndices(
+    const TensorView* tv,
+    const std::vector<kir::ForLoop*>& loops,
+    const IndexFromIdGraph* index_from_id_graph) {
+  std::unique_ptr<IndexFromIdGraph> index_from_id_graph_owner;
+  if (index_from_id_graph == nullptr) {
```

Comment:
Looks like the only case we need to create `index_from_id_graph` is in `getRootIndicesWithContiguity`. 

Should we just generate it in  `getRootIndicesWithContiguity` and always assume it's given in `getRootIndices` ?

One thing I really like about this refactor is the separation of functionality into small modules, and pushing this function a bit further towards that direction would be really great.

Reply:
fixed